DNNCommunity/Dnn.CommunityForums
# ENH: rename template extensions from .txt to .ascx

### Description of PR...
Change extensions on template files from .txt to .ascx

## Changes made
- renamed existing delivered templates
- changed template migration from database during 08.00.00 upgrade
- various template loading refactoring

## How did you test these updates?  
- Tested a new install and against copy of dnncommunity.org
  - ensure that default templates are properly loaded during a new install
  - ensure that existing templates are properly upgraded from 07.00.12
  - ensure that template editor in control panel functions properly 
  - ensure basic forums functionality (post topic/reply, etc.)

## PR Template Checklist



## Please mark which issue is solved


Close #460
